### PR TITLE
refactor maya-agent monitoring code

### PR DESCRIPTION
**Pull request by utkarshmani1997**
1. Why is this change necessary ?
- Read / write latency was not coming in the metrics
- unit was not correct for the logical size and actual size

2. How does this change address the issue ?
- added variable to gather read / write latency
- added variable to gather avg read / write block count per second

3. How to verify this change ?
- Run maya-agent binary locally or in maya-apiserver container, pass the
  controllerIP via flag `-c` and then do curl localhost:9500/metrics and
  make sure volume is being consumed otherwise it will show zero in all
  the stats fields.

4. What side effects does this change have ?
- none

5. Other details
fix and improvement: [#266](https://github.com/openebs/openebs/issues/266)

<!--
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/maya-agent/app/exporter/collector/collector.go [hunk 1]
```diff
@@ -15,6 +15,16 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+const (
```
Comment (prateekpandey14) [inline cmd/maya-agent/app/exporter/collector/collector.go]: move all constants to `pkg/util/constants.go`, these constants had been used in `volume_stats.go` file too.
So better to import them.

File: cmd/maya-agent/app/exporter/collector/collector.go [hunk 2]
```diff
@@ -164,32 +209,80 @@ func (e *VolumeExporter) collect() error {
...
 	}
 
-	rIOPS, _ := strconv.ParseFloat(metrics.ReadIOPS, 64)
+	err = json.NewDecoder(resp.Body).Decode(&metrics2)
+	if err != nil {
+		log.Printf("could not decode OpenEBS Volume controller metrics: %v", err)
+		return err
+	}
+
+	iRIOPS, _ := strconv.ParseFloat(metrics1.ReadIOPS, 64)
```
Comment (prateekpandey14) [inline cmd/maya-agent/app/exporter/collector/collector.go]: Add comments here about these variables

File: cmd/maya-agent/app/exporter/collector/collector.go [hunk 3]
```diff
@@ -164,32 +209,80 @@ func (e *VolumeExporter) collect() error {
...
+	fRIOPS, _ := strconv.ParseFloat(metrics2.ReadIOPS, 64)
+	rIOPS := fRIOPS - iRIOPS
```
Comment (prateekpandey14) [inline cmd/maya-agent/app/exporter/collector/collector.go]: this operation is being used more then once, better to move in a function and use it.

File: types/v1/math.go [hunk 4]
```diff
@@ -312,3 +312,25 @@ func removeBigIntFactors(d, factor *big.Int) (result *big.Int, times int32) {
...
 	return d, times
 }
+
+// SubstractFloat64 returns a - b , where a > b.
+func SubstractFloat64(a float64, b float64) float64 {
+	return (a - b)
+}
+
+// DivideFloat64 returns the value of value divided by base.
+// divide by zero case handled before calling.
```
Comment (AmitKumarDas) [inline types/v1/math.go]: Why is the divide by zero handled somewhere else ?
I feel this function is the right place

File: types/v1/math.go [hunk 5]
```diff
@@ -312,3 +312,25 @@ func removeBigIntFactors(d, factor *big.Int) (result *big.Int, times int32) {
 	}
 	return d, times
 }
+
+// SubstractFloat64 returns a - b , where a > b.
+func SubstractFloat64(a float64, b float64) float64 {
+	return (a - b)
```
Comment (AmitKumarDas) [inline types/v1/math.go]: How do we handle the negative output ? Should negative output be treated differently ?

Comment (utkarshmani1997) [reply types/v1/math.go]: we will never get negative value, we will get either greater than zero or zero

File: types/v1/math.go (hunk 4 above)
Comment (utkarshmani1997) [reply types/v1/math.go]: I handled this case before calling this function. that was requirement. although i will move all these operations to separate functions, these are required for calculating latency. An issue [#856](https://github.com/openebs/openebs/issues/856)

**Comment by utkarshmani1997** [comment]
new changes are in the PR #176

**Pull request closed without merging**